streaming/src/main/scala/org/apache/spark/streaming/dstream/WindowedDStream.scala

```diff
@@ -62,7 +65,22 @@ class WindowedDStream[T: ClassTag](
 
   override def compute(validTime: Time): Option[RDD[T]] = {
     val currentWindow = new Interval(validTime - windowDuration + parent.slideDuration, validTime)
-    val rddsInWindow = parent.slice(currentWindow)
+    val parentRddsInWindow = parent.slice(currentWindow)
+    val rddsInWindow = _initialWindow match {
+      case None => {
+        parentRddsInWindow
+      }
+      case Some (initialWindowRDD) => {
+        if(parentRddsInWindow.length - windowBatchCount == 0) {
+          // Window is full
+          parentRddsInWindow
+        } else {
+          // Fill the remainder of the window with initial data
+          parentRddsInWindow ++ initialWindowRDD.drop(
```

Comment:
The ordering of the RDDs in the union gets messed up with this.  Prepend the initialRdds to the parentRdds